solutions/src/main/java/ru/sberbank/school/task11/Streams.java

```diff
@@ -0,0 +1,146 @@
+     * @return стрим элементов из elements
+     */
+    public static <T> Streams of(Collection<T> elements) {
+        return new Streams<>(elements);
+    }
+
+    /**
+     * Фильтрация элементов по заданному правилу. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Иными словами, этот метод оставляет в коллекции только те элементы, которые удовлетворяют условию в лямбде.
+     * <p>
+     * Аналог Stream#map(Function).
+     * <p>
+     * Intermediate операция.
+     *
+     * @param predicate - правило фильтрации элементов.
+     * @return стрим
+     */
+    public Streams<T> filter(Predicate<? super T> predicate) {
+
+        return new Streams<>(list.stream()
+                .filter(predicate)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Преобразование элементов в какие-то другие элементы. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Например, данный метод должен уметь извлекать какие-то поля из объектов исходного типа (если это возможно).
+     * <p>
+     * Аналог Stream#map.
+     * <p>
+     * Intermediate операция.
+     *
+     * @param function - правило траснформации элементов.
+     * @return стрим
+     */
+    public <R> Streams<R> transform(Function<? super T, ? extends R> function) {
+        return new Streams<>(list.stream()
+                .map(function)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Сортировка элементов. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Данный метод должен уметь сортировать элементы по заданному правилу.
+     * <p>
+     * Аналог Stream#sorted
+     * <p>
+     * Intermediate операция.
+     *
+     * @param comparator - правило сортировки элементов.
+     * @return стрим
+     */
+    public Streams<T> sorted(Comparator<? super T> comparator) {
+        return new Streams<>(list.stream()
+                .sorted(comparator)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Преобразование стрима в Map. Необходимо подобрать нужные интерфейсы для передачи в этот метод.
+     * <p>
+     * Данный метод должен уметь преобразовать стрим в Map используя keyMapper и valueMapper.
+     * <p>
+     * Terminate операция.
+     *
+     * @param keyMapper   - правило создания ключа.
+     * @param valueMapper - правило создания значения.
+     * @return Map, собранная по правилам keyMapper и valueMapper
+     */
+    public <K,V> Map<K,V> toMap(Function<? super T, ? extends K> keyMapper,
+                                Function<? super T, ? extends V> valueMapper) {
+        return list.stream()
+                .collect(Collectors.toMap(keyMapper, valueMapper));
+    }
+
+    /**
+     * Преобразование стрима в Set.
+     * <p>
+     * Данный метод должен уметь преобразовать стрим в Set. Обратите внимание, что метод sorted должен учитываться
+     * при вызове этого метода.
+     * <p>
```

Comment:
&lt;p&gt; tag should be placed immediately before the first word, with no space after.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1355657835>*